### PR TITLE
feature - home api

### PR DESCRIPTION
**Pull request by at-tungnguyen**
#### What's this PR do? Please give a technical summary `(required)`
- [x] create home api

### CODE REVIEW

File: MyApp/Model/API/Core/Api.Hot.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                "a": a
+            ]
+        }
+    }
+
+    @discardableResult
+    static func query(params: QueryParams, completion: @escaping ([Item]) -> Void) -> Request? {
+        let path = Api.Path.Hot()
+
+        return api.request(method: .get, urlString: path, parameters: params.toJson(), completion: { (result) in
```
Comment (at-daiho) [inline MyApp/Model/API/Core/Api.Hot.swift]: WARNING - Unused parameter "result" in a closure should be replaced with _.

**Comment by at-daiho** [comment]
Total linter violations in pull request: 1
Serious: 0
Warnings: 1

File: MyApp/Model/API/Core/Api.Hot.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            let items: [Item] = [Item(name: "Cơm chiên", price: 10000, likeNumber: 987, thumbnail: "https://hellotars.com/wp-content/uploads/2017/06/api-image.png"),
```
Comment (at-daiho) [inline MyApp/Model/API/Core/Api.Hot.swift]: WARNING - Underscores should be used as thousand separator in large decimal numbers.

File: MyApp/Model/API/Core/Api.Hot.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                                   Item(name: "Cơm thiêu", price: 15000, likeNumber: 9789, thumbnail: "https://hellotars.com/wp-content/uploads/2017/06/api-image.png"),
```
Comment (at-daiho) [inline MyApp/Model/API/Core/Api.Hot.swift]: WARNING - Underscores should be used as thousand separator in large decimal numbers.

Comment (at-daiho) [inline MyApp/Model/API/Core/Api.Hot.swift]: WARNING - Underscores should be used as thousand separator in large decimal numbers.

File: MyApp/Model/API/Core/Api.Hot.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                                   Item(name: "Cơm xiên", price: 14000, likeNumber: 789, thumbnail: "https://hellotars.com/wp-content/uploads/2017/06/api-image.png"),
```
Comment (at-daiho) [inline MyApp/Model/API/Core/Api.Hot.swift]: WARNING - Underscores should be used as thousand separator in large decimal numbers.

File: MyApp/Model/API/Core/Api.Hot.swift [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                                   Item(name: "Cơm tiên", price: 30000, likeNumber: 1234, thumbnail: "https://hellotars.com/wp-content/uploads/2017/06/api-image.png"),
```
Comment (at-daiho) [inline MyApp/Model/API/Core/Api.Hot.swift]: WARNING - Underscores should be used as thousand separator in large decimal numbers.

Comment (at-daiho) [inline MyApp/Model/API/Core/Api.Hot.swift]: WARNING - Underscores should be used as thousand separator in large decimal numbers.

File: MyApp/Model/API/Core/Api.Hot.swift [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                                   Item(name: "Cơm điên", price: 18000, likeNumber: 2234, thumbnail: "https://hellotars.com/wp-content/uploads/2017/06/api-image.png")]
```
Comment (at-daiho) [inline MyApp/Model/API/Core/Api.Hot.swift]: WARNING - Underscores should be used as thousand separator in large decimal numbers.

Comment (at-daiho) [inline MyApp/Model/API/Core/Api.Hot.swift]: WARNING - Underscores should be used as thousand separator in large decimal numbers.

**Comment by at-daiho** [comment]
Total linter violations in pull request: 9
Serious: 0
Warnings: 9

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/Model/API/Core/Api.Hot.swift [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+//
+//  Created by asiantech on 8/9/17.
+//  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+import ObjectMapper
+
+extension Api.Hot {
```
Comment (at-linhvo) [inline MyApp/Model/API/Core/Api.Hot.swift]: fix name: `Api.Item`

File: MyApp/Model/Schema/Item.swift [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import Foundation
+import ObjectMapper
+
+final class Item: Mappable {
+
+    // MARK: - Properties
+    var name: String = ""
+    var price: Int = 0
+    var likeNumber: Int = 0
+    var thumbnail: String = ""
```
Comment (at-linhvo) [inline MyApp/Model/Schema/Item.swift]: var name = ""
var price = 0
var likeNumber = 0
var thumbnail = ""

File: MyApp/View/Controllers/Home/Drink/DrinkViewController.swift [hunk 9]
```diff
@@ -14,18 +14,41 @@ class DrinkViewController: ViewController {
     // MARK: - Properties
     @IBOutlet weak var collectionView: UICollectionView!
     let inset = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
-
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Home/Drink/DrinkViewController.swift]: insert line space

File: MyApp/View/Controllers/Home/Drink/DrinkViewController.swift [hunk 10]
```diff
@@ -14,18 +14,41 @@ class DrinkViewController: ViewController {
...
+    // MARK: - Private
     private func setupUI() {
         collectionView.register(ItemCollectionViewCell.self)
         collectionView.dataSource = self
         collectionView.delegate = self
     }
+
+    private func setupData() {
+        viewModel.getItem {
+            self.collectionView.reloadData()
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Home/Drink/DrinkViewController.swift]: add `[weak self] in`

File: MyApp/View/Controllers/Home/Food/FoodViewController.swift [hunk 11]
```diff
@@ -14,18 +14,42 @@ class FoodViewController: ViewController {
...
+    // MARK: - Private
     private func setupUI() {
         collectionView.register(ItemCollectionViewCell.self)
         collectionView.dataSource = self
         collectionView.delegate = self
     }
+
+    private func setupData() {
+        viewModel.getItem {
+            self.collectionView.reloadData()
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Home/Food/FoodViewController.swift]: add `[weak self] in`

File: MyApp/View/Controllers/Home/Hot/HotViewController.swift [hunk 12]
```diff
@@ -14,18 +14,42 @@ class HotViewController: ViewController {
...
+    // MARK: - Private
     private func setupUI() {
         collectionView.register(ItemCollectionViewCell.self)
         collectionView.dataSource = self
         collectionView.delegate = self
     }
+
+    private func setupData() {
+        viewModel.getItem {
+            self.collectionView.reloadData()
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Home/Hot/HotViewController.swift]: add `[weak self] in`

**Pull request merged**